src/tests/tables.test.js

```diff
@@ -20,3 +20,60 @@ describe("duplicate indices", () => {
     expect(errors).toEqual([]);
   });
 });
+
+describe("broken links", () => {
+  it("should not contain broken links", () => {
+    let errors = [];
+    let files = glob.sync("src/*.json");
+
+    let resources = {};
+
+    const walk_recursive = (resources, file, object, errors) => {
+      for (const property in object) {
+        if (property === "url") {
+          if (resources[object.url] === undefined) {
+            errors.push(`${file}: URL '${object.url}' not found.`);
+          } else {
+            if (resources[object.url].name !== undefined && resources[object.url].name !== object.name) {
+              errors.push(`${file}: Name mismatch for reference to '${object.url}', '${object.name}' should be '${resources[object.url].name}'`);
+            }
+
+            if (object.index !== undefined && resources[object.url].index !== object.index) {
+              errors.push(`${file}: Index mismatch for reference to '${object.url}', '${object.index}' should be '${resources[object.url].index}'`);
+            }
+          }
```

Comment:
You can probably just extract this out of the `for` loop since you just want to know if the object has the `url` key. Like this:
```js
if (object.hasOwnProperty('url') {
```

That will clean up your `for` loop to only be looking for subobjects.